contrib/pg_tde/src/catalog/tde_principal_key.c

```diff
@@ -813,26 +871,81 @@ GetPrincipalKey(Oid dbOid, LWLockMode lockMode)
 	return get_principal_key_from_keyring(dbOid, true);
 }
 
+TDEPrincipalKey *
+GetPrincipalKey(Oid dbOid, LWLockMode lockMode)
+{
+	TDEPrincipalKey *principalKey = GetPrincipalKeyNoDefault(dbOid, lockMode);
+#ifndef FRONTEND
+	TDEPrincipalKey *newPrincipalKey = NULL;
+#endif
```

Comment:
Wouldn't it be cleaner to just ifdef the whole body?

```
#ifdef FRONTEND
    return GetPrincipalKeyNoDefault(dbOid, lockMode);
#else
    ...
#endif
```

Reply:
Maybe, but this follows how most functions have ifdefs in the code.